### ISSUE TITLE
Lexorank Collision

### ISSUE THREAD

**Issue by zakinadhif** [DennisSnijder/payload-workflow#4]
Hello!

I've been trying to replicate some of the mechanisms which are used here for my project. I've found which appears to be a strange behavior on the `generateOrderRank` hook, and I'm not sure if this is the expected output.

Basically, it appears that the hook gets stuck and generates the same lexorank multiple times because it received an input from a wrongly sorted query?

![image](https://github.com/DennisSnijder/payload-workflow/assets/56494343/339b2a5d-2e4e-4965-a727-bd701e9357b1)

```js
  const collectionData = await payload.find({
    collection: 'menu-items',
    sort: 'orderRank',
    limit: 1,
  });

  const lastOrderRank = collectionData.docs[0]?.orderRank ?? null;

  console.log("Last order rank:", lastOrderRank)

  const lastRank =
    lastOrderRank && typeof lastOrderRank === 'string'
      ? LexoRank.parse(lastOrderRank)
      : LexoRank.min()

  console.log("Last rank:", lastRank.toString());

  const nextRank = lastRank.genNext().genNext();
  data.orderRank = nextRank.toString();

  console.log("Next rank:", nextRank.toString());
```

![image](https://github.com/DennisSnijder/payload-workflow/assets/56494343/9fbb6177-95a8-475c-90a4-909dc9d7a19b)

I suspect it might be happening here too, since my code is nearly identical since I copied from here. The problem disappears when I flip the order of the sorting as such:

```js
  sort: '-orderRank'
```

Thanks for making the plugin! I'll be using this as a reference on making my custom tree collection.

**Comment by DennisSnijder** [comment]
Interesting find! I'll take a look later this week! Thanks for the report!

**Comment by DennisSnijder** [comment]
@zakinadhif Thank you for looking into this! This bug actually resolved into the user being unable to sort their documents in some cases. This resolved that! 🥳 

The fix is available in 0.3.2
